firestore/src/android/firestore_android.cc

```diff
@@ -515,6 +542,58 @@ void FirestoreInternal::ClearListeners() {
   listener_registrations_.clear();
 }
 
+Future<LoadBundleTaskProgress> FirestoreInternal::LoadBundle(
+    const std::string& bundle) {
+  Env env = GetEnv();
+  Local<LoadBundleTaskInternal> task = CreateLoadBundleTask(env, obj_, bundle);
+  return promises_->NewFuture<LoadBundleTaskProgress>(env, AsyncFn::kLoadBundle,
+                                                      task);
+}
+
+Future<LoadBundleTaskProgress> FirestoreInternal::LoadBundle(
+    const std::string& bundle,
+    std::function<void(const LoadBundleTaskProgress&)> progress_callback) {
+  Env env = GetEnv();
+  Local<LoadBundleTaskInternal> task = CreateLoadBundleTask(env, obj_, bundle);
+
+  auto* listener = new LambdaEventListener<LoadBundleTaskProgress>(
+      [progress_callback](const LoadBundleTaskProgress& p, Error e,
+                          const std::string& s) {
+        // `e` is always `kErrorOk` and `s` is always empty for progress
+        // listeners.
+        progress_callback(p);
+      });
+  // TODO(b/187420421): The listener is owned by the firestore instance, longer
```

Comment:
Here is the bug tracking the feature to which @wu-hui alluded: b/187420421

Reply:
Thanks for the explanation!

> Unique pointer is here to help these listener to be owned by the firestore instance, which make sure they only get deleted when the firestore instance is going away.

What I'm trying to figure out is, what's the advantage of using unique pointers instead of just plain values?

Reply:
Changed to value.